tests/unit/scripting.tcl

```diff
@@ -616,6 +616,49 @@ start_server {tags {"scripting"}} {
              [run_script {return redis.sha1hex('Pizza & Mandolino')} 0]
     } {da39a3ee5e6b4b0d3255bfef95601890afd80709 74822d82031af7493c20eefa13bd07ec4fada82f}
 
+    test "redis.monotonic() measures elapsed time" {
+        set escaped [run_script {
+            local start = redis.monotonic()
+            while redis.monotonic() - start < 1000000 do end
+            return {double = redis.monotonic() - start}
+        } 0]
+        assert_morethan_equal $escaped 1000000 ;# 1 second (1 second = 1000000 microseconds)
+    }
+
+    test "os.clock() measures elapsed time" {
+        set escaped [run_script {
+            local start = os.clock()
+            while os.clock() - start < 1 do end
+            return {double = os.clock() - start}
+        } 0]
+        assert_morethan_equal $escaped 1 ;# 1 second
+    }
+
+    test "Prohibited dangerous lua methods in sandbox" {
+        assert_equal "" [run_script {
+            local allowed_methods = {"clock"}
+            -- Find a value from a tuple and return the position.
+            local indexOf = function(tuple, value)
+                for i, v in ipairs(tuple) do
+                    if v == value then return i end
+                end
+                return nil
+            end
+            -- Check for disallowed methods and verify all allowed methods exist.
+            -- If an allowed method is found, it's removed from 'allowed_functions'.
+            -- If 'allowed_functions' is empty at the end, all allowed methods were found.
+            for key, value in pairs(os) do
+                if type(value) == "function" then
```

Comment:
Why not checking all the types? Why just functions?

Reply:
Also, should we check that `ipairs` on `os` returns no values?

Reply:
I wonder if we should extend the `luaSetAllowListProtection` mechanism to allow to also have a nested allow lists for each key?

Reply:
`if #allowed_methods ~= 0` verifies that `clock` was actually found on `os`. I am asking what if accidently we will add an unwanted value inside the table index (not a key), I do not think we verify that.

Reply:
> Do you mean following structure?
os(table)
> - another_table
> - a method
> - b method
IMHO, this should be rare, and adding new lua methods should be done very carefully.

I mean that what if there are other keys which are not a function, we do not want them right? In the current code we will skip them, no?

Reply:
@MeirShpilraien please have a look the commit https://github.com/redis/redis/pull/12971/commits/84526fabc2ab9de72733aadd11edf7aeb06b7678

Reply:
Looks good :+1: